components/omnibox/browser/search_suggestions/search_suggestions_query_check_utils.cc

```diff
@@ -0,0 +1,342 @@
+    url = try_parse(trunc_query);
+    if (url.is_valid()) {
+      return url;
+    }
+  }
+
+  return GURL();
+}
+
+// Very crude heuristic to protect against leaking urls.
+// Assumes that real URLs, starting with "http[s]://"
+// have been already filtered. The rough idea is to
+// try whether "http://<query>" is a valid URL and whether
+// it has enough sensitive information to block the search.
+bool IsPotentiallyLeakingUrlInformation(const std::string& query) {
+  // Early exit: If the URL is too small, we can avoid the
+  // more expensive checks. This value should be quite conservative,
+  // otherwise URL shorter links may slip through
+  // (e.g., "goo.gl/bdkh1L", "t.co/RUiFUYKzkz", 'is.gd/PazNcR').
+  //
+  // Note: URL shorteners in general are a problem, as they provide
+  // an extremely compact representation of an URL. Although it is
+  // safe to assume that they do not encode URLs with secrets, we
+  // would still leak the site that the user is going to visit.
+  if (query.length() <= 11 ||
+      (query.length() <= 18 && !IsSmallQueryButCouldBeUrlShortener(query))) {
+    return false;
+  }
+
+  const auto url = TryParseAsUrlWithIncompleteSchema(query);
+  if (!url.is_valid()) {
+    // does not look like an URL --> safe
+    return false;
+  }
+
+  // reject non-public URLs
+  if (url.has_username() || url.has_password() || url.has_port()) {
+    return true;
+  }
+
+  // If the URL path is non empty, it is a strong indicator
+  // that the user is currently typing an URL:
+  //
+  // * If the path name itself gets too long, we have to be careful
+  //   because of links from URL shortener (e.g., 'bit.ly/1h0ceQI').
+  // * If it contains URL search paramters in addition to the
+  //   path ('<domain>/path?param[=key]'), also stop.
+  //
+  // Note: ".search" without ".pathname" is quite aggressive,
+  // for instance, 'Fu?ball' (misspelling for "Fußball"), would
+  // already match ("http://Fu?ball" => host: "Fu", search: "ball")
+  if (url.path() != "/" && (url.path().length() >= 6 || url.has_query())) {
+    return true;
+  }
+
+  const auto domain_guessable = RE2::PartialMatch(query, "\\w+[.]\\w+\\/");
+  if (domain_guessable) {
+    return true;
+  }
+
+  // looks safe
+  return false;
+}
+
+}  // namespace
+
+namespace search_suggestions {
+
+bool IsSuspiciousQuery(const std::string& query) {
+  if (!HasValidWordCountInQuery(query)) {
+    return true;
+  }
+
+  if (HasLongNumberInQuery(query)) {
+    return true;
+  }
+
+  if (HasEmailInQuery(query)) {
+    return true;
+  }
+
+  if (LooksLikeHttpPass(query)) {
+    return true;
+  }
+
+  if (CheckHashProb(query)) {
+    return true;
+  }
+
+  return false;
+}
+
+bool IsSafeQueryUrl(const std::string& query) {
+  // fast path, which should handle most calls
+  const auto is_small_enough = [](const std::string& query) {
+    return query.length() <= 6;
+  };
+
+  if (is_small_enough(query)) {
+    return true;
+  }
+
+  if (query.length() > 500) {
```

Comment:
I think this threshold should be 50, as per https://github.com/brave/web-discovery-project/blob/e58b8e46e062200a4c4d0fb147cb2e88536f5d43/modules/web-discovery-project/sources/web-discovery-project.es#L5022

Reply:
nevermind, i see that you have:

https://github.com/brave/brave-core/blob/6650430ac655a2c02510122b1f10d8e9d4fbf80a/components/omnibox/browser/brave_search_provider.cc#L26-L31

is this check redundant?

Reply:
Right, not necessary. deleted. Ported brave search's equivalent api but it's more general api.
It's redundant here.